File: net_examples/firebase/FirebaseSettingEx/app/src/main/java/lee/dorian/android/firebasesettingex/MainActivity.java
package lee.dorian.android.firebasesettingex;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.util.Log;
import android.widget.TextView;

import com.google.android.gms.common.ConnectionResult;
import com.google.android.gms.common.GoogleApiAvailability;
import com.google.firebase.FirebaseApp;
import com.google.firebase.iid.FirebaseInstanceId;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        FirebaseApp.initializeApp(this);
        int result = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(this);
        if (result == ConnectionResult.SUCCESS) {
            readToken();
        }
    }

    @Override
    protected void onResume() {
        super.onResume();

    }

    private void readToken() {
        FirebaseInstanceId.getInstance().getInstanceId().addOnCompleteListener(task -> {
            if (!task.isSuccessful()) {
                Log.w("MainActivity", "getInstanceId failed", task.getException());
                return;
            }

            String token = task.getResult().getToken();
            TextView tvToken = (TextView)findViewById(R.id.tvToken);
            tvToken.setText(token);
        });
    }

}

File: ui_examples/views/FontEx/ex01/src/main/java/lee/dorian/android/font_ex/ex01/MainActivity.kt
package lee.dorian.android.font_ex.ex01

import android.graphics.Typeface
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TextView

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val tvHelloWorld = findViewById(R.id.tvHelloWorld) as TextView
        val typeface = Typeface.createFromAsset(assets, "font_nanum_gothic.ttf")
        tvHelloWorld.typeface = typeface
    }
}